pyresample/bucket/__init__.py

```diff
@@ -202,20 +202,30 @@ def _get_indices(self):
         target_shape = self.target_area.shape
         self.idxs = self.y_idxs * target_shape[1] + self.x_idxs
 
-    def get_sum(self, data, skipna=True):
+    def get_sum(self, data, fill_value=np.nan, skipna=True,
+                set_empty_bucket_to=0):
         """Calculate sums for each bin with drop-in-a-bucket resampling.
 
         Parameters
         ----------
         data : Numpy or Dask array
             Data to be binned and summed.
+        fill_value : float
+            Fill value to mark missing/invalid values in the input data.
+            Default: np.nan
         skipna : boolean (optional)
-                If True, skips NaN values for the sum calculation
-                (similarly to Numpy's `nansum`). Buckets containing only NaN are set to zero.
-                If False, sets the bucket to NaN if one or more NaN values are present in the bucket
-                (similarly to Numpy's `sum`).
-                In both cases, empty buckets are set to 0.
-                Default: True
+            If True, skips missing values (as marked by NaN or `fill_value`) for the sum calculation
+            (similarly to Numpy's `nansum`). Buckets containing only missing values are set to `set_empty_bucket_to`.
+            If False, sets the bucket to fill_value if one or more missing values are present in the bucket
+            (similarly to Numpy's `sum`).
+            In both cases, empty buckets are set to `set_empty_bucket_to`.
+            Default: True
+        set_empty_bucket_to : float
```

Comment:
Can we work on a better name for this? The "set" part is a verb and seems more like the name for a function than a variable. What about `empty_fill` or something else? Also in what cases would this parameter not be 0?

Reply:
good point, thanks. `empty_fill` doesn't feel very self-explanatory to me though.. maybe `emtpy_bucket_value`?

In my use case, for example, I'm setting it to `np.nan` to maintain a sparse array after resampling, which is easier/lighter to handle and maintains transparency properties.

Reply:
enpty_bucket_value is better but "value" seems redundant. Like calling a variable "var". But I'm not sure it is that big of a deal that it *needs* changing.

Reply:
I was proposing `emtpy_bucket_value` since it goes along `fill_value`. But we can agree on `empty_bucket` :)

> But I'm not sure it is that big of a deal that it needs changing.

Do you mean the variable name or the entire concept of the new kwarg?

Reply:
I meant that `empty_bucket_value` is probably fine and good point with `fill_value` (although that might be historic with CF `_FillValue` and similar). Anyway...`empty_bucket_value` sounds good.

Reply:
Done, I also added a `if empty_bucket_value != 0:` check to avoid an unnecessary `where` call for the default case.